fog/api/proto/ledger.proto

```diff
@@ -8,6 +8,94 @@ import "fog_common.proto";
 package fog_ledger;
 option go_package = "mobilecoin/api";
 
+import "google/protobuf/empty.proto";
+
+////
+// Ledger router API
+////
+
+service LedgerAPI {
+    rpc request(stream LedgerRequest) returns (stream LedgerResponse) {}
+}
+
+service LedgerRouterAdminAPI {
+    // Adds a shard to the Fog View Router's list of shards to query.
+    rpc addShard(fog_common.AddShardRequest) returns (google.protobuf.Empty) {}
+}
+
+/// Fulfills requests sent by the Fog Ledger Router. This is not meant to fulfill requests sent directly by the client.
+service KeyImageStoreAPI {
+    /// This is called to perform IX key exchange with the enclave before calling GetOutputs.
+    rpc Auth(attest.AuthMessage) returns (attest.AuthMessage) {}
+    /// Input should be an encrypted MultiKeyImageStoreRequest, result is an encrypted response.
+    rpc MultiKeyImageStoreQuery(MultiKeyImageStoreRequest) returns (MultiKeyImageStoreResponse) {}
+}
+
+message LedgerRequest {
+    oneof request_data { 
+        attest.AuthMessage auth = 1;
+        attest.Message check_key_images = 2;
+        // TODO: Fill in block query service and merkele proof service.
+        // Potentially untrusted_tx_out_service? To be decided.     
+    }
+}
+
+message LedgerResponse {
+    oneof response_data { 
+        attest.AuthMessage auth = 1;
+        attest.Message check_key_image_response = 2;
+        // TODO: Fill in block query service and merkele proof service.
+        // Potentially untrusted_tx_out_service? To be decided.     
+    }
+}
+
+// Identical to FogViewStoreDecryptionError - TODO, determine if we should unify these types.
+message FogLedgerStoreDecryptionError {
+    /// The FogLedgerStoreUri for the specific Fog Ledger Store that
+    /// tried to decrypt the MultiKeyImageStoreRequest and failed.
+    /// The client should subsequently authenticate with the machine
+    /// described by this URI.
+    string store_uri = 1;
+
+    /// An error message that describes the decryption error.
+    string error_message = 2;
+}
+
+// Identical to MultiViewStoreQueryRequest - TODO, determine if we should unify these types.
+message MultiKeyImageStoreRequest {
+    /// A list of queries encrypted for Fog Ledger Stores.
+    repeated attest.NonceMessage queries = 1;
+}
+
+
+/// The status associated with a MultiViewStoreQueryResponse
+enum MultiKeyImageStoreResponseStatus {
```

Comment:
The FVR analog (`MultiViewStoreQueryResponseStatus`) added an `UNKNOWN = 0` value because proto enums if left unfilled will default to the `=0` value. This prevents unfilled statuses from containing a success value, which, if left unfilled, probably wasn't the case. 

I'd add this here as well and then add a match clause in  the shard response processor.

Reply:
Addressed in https://github.com/mobilecoinfoundation/mobilecoin/pull/2889/commits/c5cd8d8dec90f25a7a58474d505b2225dc38d261